File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int letter_counter(string text);
int word_counter(string text);
int sentence_counter(string text);
void coleman_liau_formula(string text);

int main(void)
{
    string text = get_string("Text: ");
    //letter_counter(text); uncommented or commented seems to both work
    //word_counter(text);
    //sentence_counter(text);
    coleman_liau_formula(text);
    return 0;
}

// letter counter
int letter_counter(string text)
{
    int letter = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]) != 0)
        {
            letter++;
        }
    }
    return letter;
}

// word counter
int word_counter(string text)
{
    int spaces = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isspace(text[i]) != 0)
        {
            spaces++;
        }
    }
    return spaces;
}

// sentence counter
int sentence_counter(string text)
{
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!'
            || text[i] == '?') //one flaw is that an occurence such as ellipsis would increase the sentence counter
        {
            sentences++;
        }
    }
    return sentences;
}

void coleman_liau_formula(string text)
{
    float L = ((float) letter_counter(text) / (float) word_counter(text)) * 100;
    float S = ((float) sentence_counter(text) / (float) word_counter(text)) * 100;
    int index = round((0.0588 * L) - (0.296 * S) - 15.8);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: pset6/readability/readability.py
from cs50 import get_string


def ask_for_text():
    text = get_string("Text: ")
    coleman_liau_index(text)


def letter_counter(text):
    counter = 0
    for letters in text:
        if letters.isalpha() == True:
            counter += 1
    return counter


def word_counter(text):
    words = len(text.split())  # splits words into a list
    return words


def sentence_counter(text):
    return text.count('.') + text.count('?') + text.count('!')  # counts number of respective punctuation


def coleman_liau_index(text):
    L = (letter_counter(text) / word_counter(text)) * 100
    S = (sentence_counter(text) / word_counter(text)) * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


ask_for_text()